src/_pytest/recwarn.py

```diff
@@ -206,12 +206,23 @@ def __len__(self) -> int:
         return len(self._list)
 
     def pop(self, cls: Type[Warning] = Warning) -> "warnings.WarningMessage":
-        """Pop the first recorded warning, raise exception if not exists."""
+        """Pop the first recorded warning (or subclass of warning), raise exception if not exists."""
```

Comment:
I think we'll want to give a bit more detail here, and possibly an example.  Something like

```suggestion
        """Pop the first recorded warning which is an instance of ``cls``, but not an instance 
        of a child class of any other match.  Raises ``AssertionError`` if there is no match.
        """
```